src/vs/workbench/contrib/customEditor/common/customTextEditorModel.ts

```diff
@@ -58,12 +64,39 @@ export class CustomTextEditorModel extends Disposable implements ICustomEditorMo
 				this._onDidChangeContent.fire();
 			}
 		}));
+
+		this._register(extensionService.onWillStop(e => {
+			if (!this.isDirty()) {
+				return;
+			}
+
+			const reason = e.auto
+				? localize('vetoAutoExtHostRestart', "One of the opened editors is a custom text editor.")
+				: localize('vetoExtHostRestart', "Custom text editor '{0}' could not be saved.", this.resource.path);
+
+			e.veto((async () => {
+				if (e.auto) {
+					return true;
+				}
+
+				const didSave = await this.saveCustomEditor();
+				if (!didSave) {
+					// Veto
+					return true;
+				}
+				return false; // Don't veto
+			})(), reason);
+		}));
```

Comment:
I also feel the same way about custom text editor model like custom editor model